### PR TITLE
sp-runtime-interface-test: Fix flaky test

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: primitives/runtime-interface/test/src/lib.rs [hunk 1]
```diff
@@ -176,62 +176,85 @@ fn test_versionining_register_only() {
...
+			.output()
+			.unwrap();
+
+		assert!(output.status.success());
+		Some(output)
+	}
+}
+
 #[test]
 fn test_tracing() {
```
Comment (ggwpez) [inline primitives/runtime-interface/test/src/lib.rs:199]: This one was flaky?

Comment (bkchr) [reply primitives/runtime-interface/test/src/lib.rs:199]: ` while cargo test --release -p sp-runtime-interface-test; do true; done`

**Review by ggwpez — APPROVED** [review]
Ran it 5 times in parallel on Thread 1 with 1000 reps and no failure. But also no on master…

**Comment by michalkucharczyk** [comment]
> Why the test was failing if run in parallel with other tests within single process? Was `tracing::subscriber::default` overwritten by other test?

It should be `thread_local`, hm, idk...

**Comment by ggwpez** [comment]
> It should be `thread_local`, hm, idk...

Yes that seems to cause it. Running on one thread is fine, but on multiple it errors:  
`while cargo test --offline --release -p sp-runtime-interface-test -- --test-threads 1; do true; done`
vs
`while cargo test --offline --release -p sp-runtime-interface-test -- --test-threads 10; do true; done`
So basically the opposite of https://github.com/paritytech/substrate/issues/10479 :laughing:

**Comment by michalkucharczyk** [comment]
I played with this a little. Seems that duplicating this line:
```
let _guard = tracing::subscriber::set_default(subscriber.clone());
```
works around the problem. The span also always contains 4 elements:
```
    "free_version_1",                              
    "blake2_256_version_1",
    "return_input_version_1",                                                                                          
    "malloc_version_1",
```
Without duplication the content seems to be random.

Does it look like some race-condition in registering the subscriber? If it is then we have hidden problem somewhere. Maybe not important, idk...

**Comment by michalkucharczyk** [comment]
One more observation: on my local computer only this pair of tests is causing failures (50%):
```
cargo test -- test_return_data test_tracing
```

**Comment by bkchr** [comment]
> Does it look like some race-condition in registering the subscriber?

Yes this is the problem. I should have made the comment more "extended"
```
// Run in a different process to ensure that the `Span` is registered with our local
// `TracingSubscriber`.
```

This stuff is already tracked upstream https://github.com/tokio-rs/tracing/issues/2436 https://github.com/tokio-rs/tracing/issues/2411

We register the subcriber in a node early enough, at least I hope :D https://github.com/paritytech/substrate/blob/53f5bef318c563f3aa27b55e8c46a82e8e3d961f/client/cli/src/config.rs#L632-L649

**Pull request merged**